pkgs/development/libraries/qt-5/modules/qtwebengine.nix

```diff
@@ -56,6 +57,9 @@ qtModule {
       # TODO: be more precise
       patchShebangs .
     )
+  '' + lib.optionalString (!isNull catapult) ''
```

Comment:
Not a fan of relying on packages variables being null.

Reply:
One alternative would be adding a `postPatch` argument in `qtwebengine.nix` that's appended to the existing `postPatch` string.

Reply:
@SuperSandro2000 Would you prefer that?

Reply:
What's the problem with handling dependency-injected dependencies being null?

Reply:
One problem is that adding a package called catapult to Nixpkgs would break the build of e.g. `libsForQt514.qtwebengine`. Maybe I should explicitly set `catapult = null` there.

Reply:
Right, forgot about this, coupled with the way we have nested set this is a problem.

Maybe somehow signal the intent in the name of the parameter? In what situations is this being set to null? I don't see that being docuemnted.

If this is for "bootstrapping" the build, a parameter for `buildWithCatapult ? true` with `false` being set when bootstrapping would explain why. Otherwise some name explaining the use-case.

If this is a python2/python3 difference, could this be handled by looking at the python interpreter to discern the reason? (And with a comment explaining why.)

Reply:
I decided to just add a `postPatch` argument to the expression for qtwebengine. It's the most generic solution.

Reply:
I still think we should just rip off the band-aid now and mark it broken on pre-5.15. https://github.com/NixOS/nixpkgs/issues/201765 is basically halfway done, let's not introduce extra complexity to accomodate something that will be gone in a week anyway.

Reply:
It's really not a lot of extra complexity.

Reply:
I'll leave it up to you.

Reply:
> @SuperSandro2000 Would you prefer that?

I like the sound of that a lot more. :)

> What's the problem with handling dependency-injected dependencies being null?

It can lead to strange bugs and quirks like not noticing when it is being not used in *buildInputs or not being able to access attributes of it. It is just so much easier to assume a package is always defined even if unsupported on the platform.